sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -1012,4 +1014,182 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
       }
     }
   }
+
+  private def checkDisableBucketedScan(
+    query: String,
+    expectedNumScanWithAutoScanEnabled: Int,
+    expectedNumScanWithAutoScanDisabled: Int): Unit = {
+
+    def checkNumBucketedScan(query: String, expectedNumBucketedScan: Int): Unit = {
+      val plan = sql(query).queryExecution.executedPlan
+      val bucketedScan = plan.collect { case s: FileSourceScanExec if s.bucketedScan => s }
+      assert(bucketedScan.length == expectedNumBucketedScan)
+    }
+
+    withSQLConf(SQLConf.AUTO_BUCKETED_SCAN_ENABLED.key -> "true") {
+      checkNumBucketedScan(query, expectedNumScanWithAutoScanEnabled)
+      val result = sql(query).collect()
+
+      withSQLConf(SQLConf.AUTO_BUCKETED_SCAN_ENABLED.key -> "false") {
+        checkNumBucketedScan(query, expectedNumScanWithAutoScanDisabled)
+        checkAnswer(sql(query), result)
+      }
+    }
+  }
+
+  test("SPARK-32859: disable unnecessary bucketed table scan - basic test") {
```

Comment:
Can we move the new tests to a new suite? They are more about bucket scan planning, instead of bucketed read.

Reply:
@cloud-fan - sure, moved to a separate test suite - `DisableUnnecessaryBucketedScanSuite`.